library/src/main/java/ru/noties/markwon/spans/SpannableTheme.java

```diff
@@ -368,8 +378,16 @@ public int getCodeBackgroundColor(@NonNull Paint paint, boolean multiline) {
     }
 
     public void applyHeadingTextStyle(@NonNull Paint paint, @IntRange(from = 1, to = 6) int level) {
-        paint.setFakeBoldText(true);
-        paint.setTextSize(paint.getTextSize() * HEADING_SIZES[level - 1]);
+        if (headingTypeface == null) {
+            paint.setFakeBoldText(true);
+        } else {
+            paint.setTypeface(headingTypeface);
+        }
+        if (headingTextSizeMultipliers != null && headingTextSizeMultipliers.length >= level) {
+            paint.setTextSize(paint.getTextSize() * headingTextSizeMultipliers[level - 1]);
+        } else {
```

Comment:
BTW, maybe it will be better to remove reference to the `HEADING_SIZES` from here. For example, in the `Builder#build` we check if `headingTextSizeMultipliers` is supplied _and_ if not -> create a copy of `HEADING_SIZES` (`Arrays.copyOf()` or whatever). This way, we will have here a simple check: 

```java
if (headingTextSizeMultipliers.length >= level) {
    paint.setTextSize(paint.getTextSize() * headingTextSizeMultipliers[level - 1]);
}
```

Of cause, we could also initialize Builder with copy of `HEADING_SIZES`. Which might be more intuitive.

Reply:
Seems a little silly to copy the array in the golden path case.  Most users of this library won't need to customize the header sizes, right?

Reply:
Yeah, but the thing is we are not checking for length of default array before accessing it. I know it's a bit too defensive, but still it would be a good thing to execute the check for it also. For example, we could something like this:

```java
final float[] textSizes = headingTextSizeMultipliers != null
    ? headingTextSizeMultipliers
    : HEADING_SIZES;

if (textSizes != null && textSizes.length >= level) {
    // the code
} else {
    throw new IllegalStateException(String.format(
        Locale.US,
        "Supplied heading level: %d is invalid, where configured heading sizes are: `%s`",
        level, Arrays.toString(sizes)));
}
```

So, this way we could skip the unwanted copying of default `HEADING_SIZES` array

Reply:
Done